Payara-1601 enhanced prop subst

OndrejM opened the pull request:
Preview: https://docs.payara.fish/v/6f80902735cd22d1ed9f644eaf014a854c0f77ff/documentation/core-documentation/configuration/var-substitution.html

Feel free to edit the branch directly and commit if you want to correct minor issues.

OndrejM commented:
@mikecroft @fturizo @MattGill98 Would you please review this PR?

mikecroft commented:
Is it time to get rid of the Core Documentation/Extended Documentation distinction?

I would do the following:

* Create a new APIs parent section, under which the EJB, JBatch, JCache, JACC, Javamail stuff would all go
* Add the other sections to "Extended Documentation" + alphabetise the sections
* Rename "Extended Documentation" to "Payara Server Documentation"

Thoughts?

mikecroft commented on documentation/core-documentation/configuration/var-substitution/usage-of-variables.adoc:
```diff
@@ -0,0 +1,61 @@
+      className="com.mysql.jdbc.jdbc2.optional.MysqlXADataSource",
+      user="${mysql.user}",
+      password="${ALIAS=mysql-db-password}",
+      databaseName="mysql",
```
It would be good to include `${ENV=VAR}` syntax here too, like:

```java
databaseName="${ENV=dbName}",
```

mikecroft commented on documentation/core-documentation/configuration/var-substitution/usage-of-variables.adoc:
```diff
@@ -0,0 +1,61 @@
+*   domain.xml (inside attribute values or inside text nodes that contain a configuration value)
+*   logging.properties (inside property values)
+
+Variable references can be used inside text values also in the following places within a deployed package (_Since Payara Server 4.1.1.172_):
```
Wasn't the convention for "Since..." lines to be used as subheadings? I.e. to make it more obvious that we were dicussing Feature X and that was available from version Y?

Either way, this should be 4.1.2.172

mikecroft commented on documentation/core-documentation/configuration/var-substitution/usage-of-variables.adoc:
```diff
@@ -0,0 +1,61 @@
+
+[[how-to-use-variable-references]]
+= How to use variable references
```
If this is a "how-to", it should probably be in the User Guides section

OndrejM replied:
No, it's definitely documentation, it only covers facts and no step-by-step guide. I'll rename the header to "Using variable references".

mikecroft commented on documentation/core-documentation/configuration/var-substitution/types-of-variables.adoc:
```diff
@@ -0,0 +1,56 @@
+[[references-to-environment-variables]]
+== References to environment variables
+
+_Since Payara Server 4.1.1.171_
```
4.1.2.172

OndrejM commented:
@mikecroft The changes to structure you proposed are fine. I would do them after we merge all important 172 stuff into next and then to master. This PR is important to get into the current docs ASAP because the functionality was requested by a customer.

OndrejM commented:
@mikecroft I updated the docs according to your comments. I split the usage into 2 sections, the 2nd section is only since 172.

Preview: https://docs.payara.fish/v/fc47058e5030d0cc2e347eb8a28363fe88dbfa26/documentation/core-documentation/configuration/var-substitution/usage-of-variables.html

OndrejM commented:
@mikecroft Can you review this? I'd like to merge it and inform customers.
